### ISSUE TITLE
Consider switching global-tunnel-ng to global-agent

### ISSUE THREAD

**Issue by gajus** [cyclosproject/ng-swagger-gen#189]
`global-tunnel-ng` used in this package [does not work in Node v11 and above](https://github.com/np-maintain/global-tunnel#global-tunnel).

If you require a global proxy, consider switching to https://github.com/gajus/global-agent.

The required setup is:

```js
import {
  createGlobalProxyAgent
} from 'global-agent';

createGlobalProxyAgent({
  environmentVariableNamespace: '',
  forceGlobalAgent: true
});

```

**Comment by Jejuni** [comment]
Strongly support this.
Ran into the issue today and arrived at the conclusion that the combination of our corporate proxy and `global-tunnel-ng` caused the `Cannot read property 'proxy' of undefined` error message.

**Comment by luisfpg** [comment]
I don't use a proxy myself.
Would this solution from https://github.com/gajus/global-agent#setup-proxy-using-bootstrap-routine be acceptable?

```javascript
import {
  bootstrap
} from 'global-agent';
import globalTunnel from 'global-tunnel-ng';

const MAJOR_NODEJS_VERSION = parseInt(process.version.slice(1).split('.')[0], 10);

if (MAJOR_NODEJS_VERSION >= 10) {
  // `global-agent` works with Node.js v10 and above.
  bootstrap();
} else {
  // `global-tunnel-ng` works only with Node.js v10 and below.
  globalTunnel.initialize();
}
```

**Comment by gajus** [comment]
To make it fully compatible, you would need to use `environmentVariableNamespace` configuration, e.g.

```js
import {
  bootstrap
} from 'global-agent';
import globalTunnel from 'global-tunnel-ng';

const MAJOR_NODEJS_VERSION = parseInt(process.version.slice(1).split('.')[0], 10);

if (MAJOR_NODEJS_VERSION >= 10) {
  // `global-agent` works with Node.js v10 and above.
  bootstrap({
    environmentVariableNamespace: ''
  });
} else {
  // `global-tunnel-ng` works only with Node.js v10 and below.
  globalTunnel.initialize();
}

```

**Comment by luisfpg** [comment]
Can you, please, test the [proxy branch](https://github.com/cyclosproject/ng-swagger-gen/tree/proxy) to see if it works?
If ok I'll make a new release to support this.

**Comment by Jejuni** [comment]
The implementation isn't going to keep compatibility with all users, as `global-tunnel-ng` also checks npm config for proxy settings.

I created a pull request #193 that should hopefully cover all use cases.
@gajus Now that I think about it, I never checked whether `global-agent` internally already removes the HTTP_PROXY env variables. I went off of the documentation stating that the default is to use the GLOBAL_AGENT_ namespace prefix. Are you doing any processing internally to the env variables?

Otherwise, for compatibility with `global-tunnel-ng`, the processing should probably be left in.
